chgnet/graph/fast_converter_libraries/uthash.h

```diff
@@ -0,0 +1,1140 @@
+/*
```

Comment:
Asking without any background knowledge here but do we actually need to check this into version control? I know C is not as batteries-included as Python but seems strange that you even have to bring your own hash table.

Reply:
C doesn't support any data structures (unless strings are a data structure) in their standard library -- in fact, their standard library is extremely sparse compared to languages such as Java or Python. This is a useful hashmap I found, implemented purely in C macros, on the interwebs.

Reply:
Do any of the C++ standard data structures work for this case?

Reply:
There might be -- I'm not familiar with C++ however. The uthash.h library is simple to use.